render-ws-spark-client/src/main/java/org/janelia/render/client/spark/match/CopyMatchClient.java

```diff
@@ -47,56 +49,63 @@ public Parameters(final MultiProjectParameters multiProject,
             this.multiProject = multiProject;
             this.matchCopy = matchCopy;
         }
-
     }
 
+    /** Run the client with command line parameters. */
     public static void main(final String[] args) {
-
         final ClientRunner clientRunner = new ClientRunner(args) {
             @Override
             public void runClient(final String[] args) throws Exception {
-
                 final Parameters parameters = new Parameters();
                 parameters.parse(args);
                 parameters.matchCopy.validate();
-
-                final CopyMatchClient client = new CopyMatchClient(parameters);
-                client.run();
-
+                final CopyMatchClient client = new CopyMatchClient();
+                client.createContextAndRun(parameters);
             }
         };
         clientRunner.run();
-
     }
 
-    private final Parameters parameters;
-
-    public CopyMatchClient(final Parameters parameters) throws IllegalArgumentException {
-        LOG.info("init: parameters={}", parameters);
-        this.parameters = parameters;
+    /** Empty constructor required for alignment pipeline steps. */
+    public CopyMatchClient() {
     }
 
-    public void run() throws IOException {
-
-        final SparkConf conf = new SparkConf().setAppName("CopyMatchClient");
-
+    /** Create a spark context and run the client with the specified parameters. */
+    public void createContextAndRun(final Parameters clientParameters) throws IOException {
+        final SparkConf conf = new SparkConf().setAppName(getClass().getSimpleName());
         try (final JavaSparkContext sparkContext = new JavaSparkContext(conf)) {
-            final String sparkAppId = sparkContext.getConf().getAppId();
-            LOG.info("run: appId is {}", sparkAppId);
-            copyMatches(sparkContext);
+            LOG.info("createContextAndRun: appId is {}", sparkContext.getConf().getAppId());
+            copyMatches(sparkContext, clientParameters);
         }
     }
 
-    public void copyMatches(final JavaSparkContext sparkContext)
-            throws IOException {
+    /** Validates the specified pipeline parameters are sufficient. */
+    @Override
+    public void validatePipelineParameters(final AlignmentPipelineParameters pipelineParameters)
+            throws IllegalArgumentException {
+        AlignmentPipelineParameters.validateRequiredElementExists("matchCopy",
+                                                                  pipelineParameters.getMatchCopy());
+    }
```

Comment:
The validation seems to only verify that the parameters are present, not that they are valid. Is the name maybe too general for the purpose it serves?

Reply:
My intent/hope is that we flesh-out the validation implementations to do more later - assuming that is useful/helpful.